### PR TITLE
Code splitting

### PR DESCRIPTION
**Pull request by sourabhmodi20051**
Configure Redux with dynamic reducer injection
Added sample usage of redux in app using hooks

### CODE REVIEW

**Comment by nimish-gupta** [comment]
Why are you using redux?

**Comment by sourabhmodi20051** [comment]
> Why are you using redux?

Every message send and receive will change UI in multiple components like - Message box, count of replies in the thread, notification count, last message in the conversation list, etc.  It will be easy to maintain this state using redux.

**Comment by nimish-gupta** [comment]
> > Why are you using redux?
> 
> Every message send and receive will change UI in multiple components like - Message box, count of replies in the thread, notification count, last message in the conversation list, etc. It will be easy to maintain this state using redux.

React context?

**Comment by sourabhmodi20051** [comment]
> > > Why are you using redux?
> > 
> > 
> > Every message send and receive will change UI in multiple components like - Message box, count of replies in the thread, notification count, last message in the conversation list, etc. It will be easy to maintain this state using redux.
> 
> React context?

The context will share state in the tree, in messaging case we will have to share state with sibling components as well.

**Comment by nimish-gupta** [comment]
F

> > > > Why are you using redux?
> > > 
> > > 
> > > Every message send and receive will change UI in multiple components like - Message box, count of replies in the thread, notification count, last message in the conversation list, etc. It will be easy to maintain this state using redux.
> > 
> > 
> > React context?
> 
> The context will share state in the tree, in messaging case we will have to share state with sibling components as well.

For sibling components, you can pass the state to the parent component. You want to share the state into tree, right? 
For sibling components you can pass state to them individually.

**Comment by sourabhmodi20051** [comment]
> F
> 
> > > > > Why are you using redux?
> > > > 
> > > > 
> > > > Every message send and receive will change UI in multiple components like - Message box, count of replies in the thread, notification count, last message in the conversation list, etc. It will be easy to maintain this state using redux.
> > > 
> > > 
> > > React context?
> > 
> > 
> > The context will share state in the tree, in messaging case we will have to share state with sibling components as well.
> 
> For sibling components, you can pass the state to the parent component. You want to share the state into tree, right?
> For sibling components you can pass state to them individually.

Will that not increase the code complexity? With hooks, all the heavy lifting of redux need not be done by us.

**Comment by nimish-gupta** [comment]
> > F
> > > > > > Why are you using redux?
> > > > > 
> > > > > 
> > > > > Every message send and receive will change UI in multiple components like - Message box, count of replies in the thread, notification count, last message in the conversation list, etc. It will be easy to maintain this state using redux.
> > > > 
> > > > 
> > > > React context?
> > > 
> > > 
> > > The context will share state in the tree, in messaging case we will have to share state with sibling components as well.
> > 
> > 
> > For sibling components, you can pass the state to the parent component. You want to share the state into tree, right?
> > For sibling components you can pass state to them individually.
> 
> Will that not increase the code complexity? With hooks, all the heavy lifting of redux need not be done by us.

As if now I can't comment on the code complexity. Hooks has also made the use of context api

**Comment by sourabhmodi20051** [comment]
> > > F
> > > > > > > Why are you using redux?
> > > > > > 
> > > > > > 
> > > > > > Every message send and receive will change UI in multiple components like - Message box, count of replies in the thread, notification count, last message in the conversation list, etc. It will be easy to maintain this state using redux.
> > > > > 
> > > > > 
> > > > > React context?
> > > > 
> > > > 
> > > > The context will share state in the tree, in messaging case we will have to share state with sibling components as well.
> > > 
> > > 
> > > For sibling components, you can pass the state to the parent component. You want to share the state into tree, right?
> > > For sibling components you can pass state to them individually.
> > 
> > 
> > Will that not increase the code complexity? With hooks, all the heavy lifting of redux need not be done by us.
> 
> As if now I can't comment on the code complexity. Hooks has also made the use of context api

I went through Context API properly, I think redux is not required,  removed redux and added react context :)

File: client/src/containers/UserStatus/UserStatus.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React, { useContext } from "react";
+import { AppContext } from "../App/AppContext";
+
+function UserStatus() {
```
[the remark this answers is not in the document]
Comment (sourabhmodi20051) [reply client/src/containers/UserStatus/UserStatus.js]: Added Tests

**Pull request merged**